db/sql/sql.go

```diff
@@ -932,6 +998,72 @@ func (tx *Transaction) ListContext(ctx context.Context, s *schema.Schema, filter
 	return tx.executeSelect(ctx, sc, sql, args)
 }
 
+func (tx *CachedTransaction) List(s *schema.Schema, filter transaction.Filter, options *transaction.ViewOptions, pg *pagination.Paginator) (list []*schema.Resource, total uint64, err error) {
+	return tx.ListContext(context.Background(), s, filter, options, pg)
+}
+
+//List resources in the db
+func (tx *CachedTransaction) ListContext(ctx context.Context, s *schema.Schema, filter transaction.Filter, options *transaction.ViewOptions, pg *pagination.Paginator) (list []*schema.Resource, total uint64, err error) {
+	sc := &selectContext{
+		schema:    s,
+		filter:    filter,
+		join:      true,
+		paginator: pg,
+	}
+	if options != nil {
+		sc.fields = normFields(options.Fields, s)
+		sc.join = options.Details
+	}
+
+	sql, args, err := buildSelect(sc)
```

Comment:
This probably brings performance impact since all SQL queries are generated twice. This operation requires memory allocations and iterations. Also flow is very frequent. I think we should calculate hash for parameters of TxTransaction methods, not SQL query.

Reply:
I did it because i wasn't sure if there could be any changes in schema that could cause generating different query for the same parameters. But i can investigate it further